File: frontend/src/types/project.ts
// GET レスポンス用（FastAPI の ProjectResponse に対応）
export type Project = {
  id: number;
  title: string;
  comment: string | null;
  genre: string;
  deadline: string | null;
  user_id1: number;
  user_id2: number;
  created_at: string;
};

// POST リクエスト用（FastAPI の ProjectCreate に対応）
export type ProjectCreate = {
  other_user_id: number;
  title: string;
  comment?: string | null;
  genre: string;
  deadline?: string | null;
};

File: backend/app/schemas/project.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class ProjectCreate(BaseModel):
    """POST リクエスト時に受け取るデータ"""
    other_user_id: int
    title: str
    comment: Optional[str] = None
    genre: str
    deadline: Optional[date] = None


class ProjectResponse(BaseModel):
    """レスポンスとして返すデータ"""
    id: int
    title: str
    comment: Optional[str] = None
    genre: str
    deadline: Optional[date] = None
    user_id1: int
    user_id2: int
    created_at: datetime

    class Config:
        from_attributes = True
